File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceProfileView.java
package com.sequenceiq.cloudbreak.cloud.aws.view;

import java.util.Optional;

import com.sequenceiq.cloudbreak.cloud.model.CloudStack;
import com.sequenceiq.cloudbreak.cloud.model.SpiFileSystem;
import com.sequenceiq.cloudbreak.cloud.model.filesystem.CloudS3View;

public class AwsInstanceProfileView {

    private final Optional<SpiFileSystem> cloudFileSystem;

    public AwsInstanceProfileView(CloudStack stack) {
        cloudFileSystem = stack.getFileSystem();
    }

    public boolean isInstanceProfileAvailable() {
        return cloudFileSystem.isPresent() && (cloudFileSystem.get().getCloudFileSystem() instanceof CloudS3View) && getInstanceProfile() != null;
    }

    public String getInstanceProfile() {
        return ((CloudS3View) cloudFileSystem.get().getCloudFileSystem()).getInstanceProfile();
    }

}

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsInstanceProfileView.kt
package com.sequenceiq.cloudbreak.cloud.aws.view

import org.apache.commons.lang3.StringUtils.isEmpty

import com.sequenceiq.cloudbreak.cloud.aws.AwsPlatformParameters
import com.sequenceiq.cloudbreak.api.model.InstanceProfileStrategy

class AwsInstanceProfileView(private val parameters: Map<String, String>) {

    val isEnableInstanceProfileStrategy: Boolean
        get() = parameters.containsKey(AwsPlatformParameters.INSTANCE_PROFILE_STRATEGY) && !isEmpty(parameters[AwsPlatformParameters.INSTANCE_PROFILE_STRATEGY].toString())

    val isS3RoleAvailable: Boolean
        get() = parameters.containsKey(AwsPlatformParameters.S3_ROLE) && !isEmpty(parameters[AwsPlatformParameters.S3_ROLE].toString())

    val s3Role: String
        get() = parameters[AwsPlatformParameters.S3_ROLE].toString()

    val instanceProfileStrategy: InstanceProfileStrategy?
        get() {
            val instanceProfileStrategy = parameters[AwsPlatformParameters.INSTANCE_PROFILE_STRATEGY]
            return if (instanceProfileStrategy == null) null else InstanceProfileStrategy.valueOf(instanceProfileStrategy)
        }

    val isCreateInstanceProfile: Boolean
        get() = InstanceProfileStrategy.CREATE == instanceProfileStrategy

    val isUseExistingInstanceProfile: Boolean
        get() = InstanceProfileStrategy.USE_EXISTING == instanceProfileStrategy

}
